lessons/08-mcp-advanced/README.md

```diff
@@ -0,0 +1,268 @@
+# Lesson 8: MCP Part II
+
+In our previous lesson, we learned how to create an MCP Server and how this is a good way to separate an AI app from its capabilities. We saw how we could add capabilities like tools and resources. Additionally, we showed how the server could be consumed by either an inspector tool or a written client. We've only scratched the surface of what MCP can do, and in this chapter we'll add an LLM to the client so you can see how this creates a better user experience.
+
+In this chapter you will learn to:
+
+- Augment your client with an LLM.
+- Use your improved client to convert an MCP Server response to a tool.
+- Leverage your improved client to create a more natural user interaction.
+
+## Set up
+
+If you haven't already, set up your development environment. Here's how you can do it: [Set up your environment](/docs/setup/README.md).
+
+## Related Resources
+
+[![Watch a short video about MCP](https://img.youtube.com/vi/YRfOiB0Im64/0.jpg)](https://www.youtube.com/watch?v=YRfOiB0Im64)
+
+_This video explains Model Context Protocol LLM usage._
+
+*🎥 Click on the image above to watch a short video about MCP*
+
+
+## Narrative: Hedwig "Hedy" Lamarr
+
+> [!NOTE] 
+> _Our story so far. You are a mechanic from 1860s London. You were working on your automaton and received a letter from Charles Babbage that ended up taking you to a library where you picked up a time travel device. Throughout your travels in time you've ended up in Florence, where you met Leonardo Da Vinci. You also went to the Aztec empire and this is where the story continues._
+>
+> See [Lesson 1](../01-intro-to-genai/README.md) if you want to catch up with the story from the beginning. 
+
+> [!NOTE] 
+> While we recommend going through the story (it's fun!), [click here](#interact-with-hedy-lamarr) if you'd prefer to jump straight to the technical content.
+
+You once again travelled back to Ada's mansion. This time Ada was meeting you at the main gates.
```

Comment:
US English versus British English but normally it's "traveled" with one L outside of the UK. :-)